cmd/api/src/api/v2/cypherquery.go

```diff
@@ -60,69 +59,69 @@ func (s Resources) CypherSearch(response http.ResponseWriter, request *http.Requ
 	}
 
 	if preparedQuery.HasMutation {
-		if !s.Authorizer.AllowsPermission(authCtx, auth.Permissions().GraphDBMutate) {
-			s.Authorizer.AuditLogUnauthorizedAccess(request)
-			api.WriteErrorResponse(
-				request.Context(),
-				api.BuildErrorResponse(http.StatusForbidden, "Permission denied: User may not modify the graph.", request), response,
-			)
-			return
-		}
-
-		// All mutation attempts must be audit logged even when failed
-		if auditLogEntry, err = model.NewAuditEntry(
-			model.AuditLogActionMutateGraph,
-			model.AuditLogStatusIntent,
-			model.AuditData{"query": preparedQuery.StrippedQuery}); err != nil {
-			api.WriteErrorResponse(
-				request.Context(),
-				api.BuildErrorResponse(http.StatusInternalServerError, api.ErrorResponseDetailsInternalServerError, request),
-				response,
-			)
-			return
-		}
+		graphResponse, err = s.cypherMutation(request, preparedQuery, payload.IncludeProperties)
+	} else {
+		graphResponse, err = s.GraphQuery.RawCypherQuery(request.Context(), preparedQuery, payload.IncludeProperties)
+	}
 
-		// create an intent audit log
-		if err := s.DB.AppendAuditLog(request.Context(), auditLogEntry); err != nil {
+	if err != nil {
+		switch true {
+		case strings.Contains(err.Error(), "Permission denied: User may not modify the graph."):
```

Comment:
I would not use a switch here, at least like this. We should definitely not look for a string in an error. Make the error a proper sentinel if we need it and use `errors.Is` to match the error we care about